src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/SerializationLogicTests.cs

```diff
@@ -39,23 +58,51 @@ public static void DictionaryFastPathPrimitiveValueSupported()
             Assert.Null(DictionaryTypeContext.Default.Int32.Serialize);
         }
 
-        // Options with features that aren't supported in generated serialization funcs.
+        // Options with features that apply only to deserialization.
+        public static IEnumerable<object[]> GetOptionsUsingDeserializationOnlyFeatures()
+        {
+            yield return new object[] { new JsonSerializerOptions(s_compatibleOptions) { AllowTrailingCommas = true } };
+            yield return new object[] { new JsonSerializerOptions(s_compatibleOptions) { DefaultBufferSize = 8192 } };
+            yield return new object[] { new JsonSerializerOptions(s_compatibleOptions) { PropertyNameCaseInsensitive = true } };
+            yield return new object[] { new JsonSerializerOptions(s_compatibleOptions) { ReadCommentHandling = JsonCommentHandling.Skip } };
+        }
+
+        /// <summary>
+        /// Options compatible with <see cref="JsonContext.s_defaultOptions"/>.
+        /// </summary>
+        public static IEnumerable<object[]> GetCompatibleOptions()
+        {
+            yield return new object[] { s_compatibleOptions };
+            yield return new object[] { new JsonSerializerOptions(JsonSerializerDefaults.Web) { DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingDefault } };
+        }
+
+        // Options with features that aren't supported in the generated serialization funcs.
         public static IEnumerable<object[]> GetOptionsUsingUnsupportedFeatures()
         {
-            yield return new object[] { new JsonSerializerOptions { Converters = { new JsonStringEnumConverter() } } };
-            yield return new object[] { new JsonSerializerOptions { Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping } };
-            yield return new object[] { new JsonSerializerOptions { NumberHandling = JsonNumberHandling.WriteAsString } };
-            yield return new object[] { new JsonSerializerOptions { ReferenceHandler = ReferenceHandler.Preserve } };
-            yield return new object[] { new JsonSerializerOptions { ReferenceHandler = ReferenceHandler.IgnoreCycles } };
-            yield return new object[] { new JsonSerializerOptions { ReferenceHandler = ReferenceHandler.IgnoreCycles } };
+            yield return new object[] { new JsonSerializerOptions(s_compatibleOptions) { Converters = { new JsonStringEnumConverter() } } };
```

Comment:
Why the change to the existing tests here? Do we no longer care about testing the scenarios that don't have `s_compatibleOptions`?

Would it be better to leave the existing tests and add these new ones as additional new cases?

Reply:
I'll move them to `GetIncompatibleOptions` below since the more pertinent issue with them (in my view) is that they don't set these two properties - https://github.com/dotnet/runtime/blob/552b8678611afb08e4743e122b917847dae6e2ce/src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/ContextClasses.cs#L35-L36.